Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -6,16 +6,60 @@
 
 #include <algorithm>
 #include <cstddef>
+#include <optional>
 #include <sstream>
 #include <string>
 #include <vector>
 
+#include <expr.h>
+
 #include "Common/CommonTypes.h"
 #include "Common/DebugInterface.h"
 #include "Common/Logging/Log.h"
 #include "Core/Core.h"
 #include "Core/PowerPC/JitInterface.h"
 #include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+struct ExprDeleter
+{
+  void operator()(expr* expression) const { expr_destroy(expression, nullptr); }
+};
+
+using ExprPointer = std::unique_ptr<expr, ExprDeleter>;
+
+static std::optional<int> ParseGPR(const char* name)
+{
+  if (name[0] == 'r' && isdigit(name[1]))
```

Comment:
Should this perhaps also perform sanity checks, such as "at least two characters long" etc?

Reply:
If you add a string length check, it might be worth changing ParseGPR to take a std::string_view so you don't have to call strlen manually.

Reply:
A length check is technically redundant, but probably not a bad idea.

Reply:
Unfortunately string_view doesn't require null termination, but strtol does. It would be unsafe to accept a string_view and pass the underlying pointer straight to strtol, so I'll leave the parameter as C string for now.

Reply:
Oh right, strtol... You can use [std::from_chars](https://en.cppreference.com/w/cpp/utility/from_chars) instead, which doesn't require a null terminated string.